cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -194,6 +197,27 @@ public KubeTwillRunnerService(MasterEnvironmentContext masterEnvContext, ApiClie
     this.workloadIdentityProvider = workloadIdentityProvider;
   }
 
+  /**
+   * Returns true if the pod is owned by a stateful set.
+   */
+  private boolean isStatefulSetPod() {
+    return this.podInfo.getOwnerReferences().size() == 1
+      && this.podInfo.getOwnerReferences().get(0).getKind().equals("StatefulSet");
+  }
+
+  /**
+   * Generates k8s label selector which selects all runs started by the k8s twill runner that has the run id label.
+   * If the pod comes from a statefulset it also adds a PARENT_POD_LABEL to the selector.
+   */
+  private String generateSelector() {
+    if (isStatefulSetPod()) {
```

Comment:
Why do we need to check stateful set only?

Reply:
As mentioned in below comment, we do not want this functionality to exist in case we are doing this in Deployment sets, because the Pod name will not be sticky identity.